### PR TITLE
Login & Register Page

### PR DESCRIPTION
**Pull request by kunaldevxxx**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
     "prepare": "husky"
   },
   "dependencies": {
+    "lucide-react": "^0.462.0",
```
Comment (abhisek) [inline package.json:13]: What is this and why are we using this?

File: src/app/globals.css [hunk 2]
```diff
@@ -2,20 +2,13 @@
...
-  --background: #ffffff;
-  --foreground: #171717;
-}
-
-@media (prefers-color-scheme: dark) {
-  :root {
-    --background: #0a0a0a;
-    --foreground: #ededed;
-  }
+@font-face {
```
Comment (abhisek) [inline src/app/globals.css:5]: Lets please keep the default fonts for now. We can create separate task for font customization. This is because font changes affect entire application experience.

File: tailwind.config.ts [hunk 3]
```diff
@@ -1,18 +1,33 @@
...
-      colors: {
-        background: "var(--background)",
-        foreground: "var(--foreground)",
+      fontFamily: {
+        geist: ['Geist', 'sans-serif'],
       },
     },
   },
   plugins: [],
-} satisfies Config;
```
Comment (abhisek) [inline tailwind.config.ts:18]: Why are we removing `satisfies Config`? This is important for type checking of Typescript

File: src/app/globals.css [hunk 4]
```diff
@@ -2,20 +2,13 @@
...
+  font-family: 'Geist';
+  src: url(../../public/fonts/GeistMonoVF.woff) format('woff');
+  font-weight: normal;
+  font-style: normal;
 }
 
 body {
-  color: var(--foreground);
```
Comment (abhisek) [inline src/app/globals.css:18]: Do we need to delete all these?

File: src/app/layout.tsx [hunk 5]
```diff
@@ -1,35 +1,17 @@
-import type { Metadata } from "next";
-import localFont from "next/font/local";
-import "./globals.css";
+import React from 'react';
+import './globals.css';
 
-const geistSans = localFont({
```
Comment (abhisek) [inline src/app/layout.tsx:5]: Why are we deleting these?

File: src/app/page.tsx [hunk 6]
```diff
@@ -1,101 +1,78 @@
-import Image from "next/image";
+import React from 'react';
+import { Lock, User, CheckCircle, Shield } from 'lucide-react';
 
-export default function Home() {
+const SecureAuthenticationUI = () => {
```
Comment (abhisek) [inline src/app/page.tsx:4]: If this is a reusable component, it should go to `src/components` Lets stick to Nextjs convention of Pages. Please see page documentation. https://nextjs.org/docs/pages/building-your-application/routing/pages-and-layouts

File: tailwind.config.ts [hunk 7]
```diff
@@ -1,18 +1,33 @@
...
+//         foreground: "var(--foreground)",
+//       },
+//     },
+//   },
+//   plugins: [],
+// } satisfies Config;
+/** @type {import('tailwindcss').Config} */
+module.exports = {
   content: [
-    "./src/pages/**/*.{js,ts,jsx,tsx,mdx}",
```
Comment (abhisek) [inline tailwind.config.ts:5]: Why are we deleting all these? We need the ability to load from `src/pages`, `src/components` etc.

File: tailwind.config.ts [hunk 8]
```diff
@@ -1,18 +1,33 @@
...
-    "./src/components/**/*.{js,ts,jsx,tsx,mdx}",
-    "./src/app/**/*.{js,ts,jsx,tsx,mdx}",
+    "./src/**/*.{js,ts,jsx,tsx}",
+    "./public/**/*.html",
```
Comment (abhisek) [inline tailwind.config.ts:23]: Why do we need to load `*.html`?

File: tailwind.config.ts [hunk 9]
```diff
@@ -1,18 +1,33 @@
...
   ],
   theme: {
     extend: {
-      colors: {
-        background: "var(--background)",
-        foreground: "var(--foreground)",
+      fontFamily: {
```
Comment (abhisek) [inline tailwind.config.ts:27]: Please keep default font. Font change can be separate task as it impacts the entire theme / application experience.

File: public/fonts/GeistMonoVF.woff
Comment (abhisek) [inline public/fonts/GeistMonoVF.woff:1]: Why are we moving this file? The Nextjs convention is to keep them in `src/app` because thats where it is required.

File: src/app/layout.tsx [hunk 10]
```diff
@@ -1,35 +1,17 @@
...
+  title: 'SafeDep Platform',
+  description: 'Safedep',
 };
 
-export default function RootLayout({
-  children,
-}: Readonly<{
-  children: React.ReactNode;
-}>) {
+const RootLayout = ({ children }: { children: React.ReactNode }) => {
```
Comment (abhisek) [inline src/app/layout.tsx:9]: Please don't change the default function signature for the `RootLayout`. It aligns with the framework default. Any specific reason for changing this?

File: src/app/page.tsx [hunk 11]
```diff
@@ -1,101 +1,78 @@
...
-        </a>
-      </footer>
-    </div>
+      </div>
+    </main>
   );
-}
+};
+
+export default SecureAuthenticationUI;
```
Comment (abhisek) [inline src/app/page.tsx:78]: Is this a page or component? If this is the home page, we should just call it `Home`. Ideally I think we should create a page called `Auth.tsx`. The `Home` page in future will check if the user is authenticated, if not, it will redirect to `Auth`, else to some other page. What do you think?

File: tailwind.config.ts [hunk 12]
```diff
@@ -1,18 +1,33 @@
-import type { Config } from "tailwindcss";
+// import type { Config } from "tailwindcss";
 
-export default {
+// export default {
```
Comment (abhisek) [inline tailwind.config.ts:3]: Do we really need to change all default Tailwind config? Lets do that in separate tasks. A single PR should ideally bring in the minimal changes required to complete the task. Additional "good to have" changes should be done in separate PRs

File: package.json (hunk 1 above)
Comment (kunaldevxxx) [reply package.json:13]: for icons sir

File: src/app/globals.css (hunk 2 above)
Comment (kunaldevxxx) [reply src/app/globals.css:5]: okay sir

File: src/app/page.tsx (hunk 11 above)
Comment (kunaldevxxx) [reply src/app/page.tsx:78]: sir its a page we can do that way also like make page and call it here but we have this initial screen so i added this in this

File: src/app/globals.css (hunk 4 above)
Comment (kunaldevxxx) [reply src/app/globals.css:18]: yes sir because we are doing our own so if we dont change so it will affect to our styling we done in code

File: src/app/layout.tsx (hunk 5 above)
Comment (kunaldevxxx) [reply src/app/layout.tsx:5]: actually i done another way its done like inside of layout font specified but i done in global css same thing   because font should be managed

Comment (abhisek) [reply src/app/layout.tsx:5]: @kunaldevxxx  Please see the Nextjs convention for handling fonts. I think they support managing fonts even when they are within app sources and need not be explicitly global

https://nextjs.org/docs/pages/building-your-application/optimizing/fonts

File: src/app/globals.css (hunk 4 above)
Comment (abhisek) [reply src/app/globals.css:18]: @kunaldevxxx This is global stylesheet. Any change here impact every single page. Can we have Auth page specific styling within the Auth page itself or just create `auth.css` and include it in `Auth.tsx` to do Auth page specific styles please? This will ensure that Auth page specific styles are not coupled with other pages.

File: tailwind.config.ts (hunk 9 above)
Comment (abhisek) [reply tailwind.config.ts:27]: @kunaldevxxx The current code is auto-generated using Nextjs cli. It follows the Nextjs framework convention. Please do not break the convention unless there is a strong reason to do so.

**Review by abhisek — APPROVED** [review]
